src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/volume/AzureVolumePlugin.java

```diff
@@ -0,0 +1,158 @@
+package cloud.fogbow.ras.core.plugins.interoperability.azure.volume;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.log4j.Logger;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.Disk;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Indexable;
+
+import cloud.fogbow.common.constants.AzureConstants;
+import cloud.fogbow.common.exceptions.FogbowException;
+import cloud.fogbow.common.exceptions.InstanceNotFoundException;
+import cloud.fogbow.common.models.AzureUser;
+import cloud.fogbow.common.util.PropertiesUtil;
+import cloud.fogbow.ras.api.http.response.InstanceState;
+import cloud.fogbow.ras.api.http.response.VolumeInstance;
+import cloud.fogbow.ras.api.http.response.quotas.allocation.VolumeAllocation;
+import cloud.fogbow.ras.constants.Messages;
+import cloud.fogbow.ras.core.models.ResourceType;
+import cloud.fogbow.ras.core.models.orders.VolumeOrder;
+import cloud.fogbow.ras.core.plugins.interoperability.VolumePlugin;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureClientCacheManager;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureGeneralUtil;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureResourceIdBuilder;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.util.AzureStateMapper;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.volume.sdk.AzureVolumeOperationSDK;
+import cloud.fogbow.ras.core.plugins.interoperability.azure.volume.sdk.AzureVolumeSDK;
+import rx.Completable;
+import rx.Observable;
+
+public class AzureVolumePlugin implements VolumePlugin<AzureUser>{
+
+    private static final Logger LOGGER = Logger.getLogger(AzureVolumePlugin.class);
+    
+    private final String defaultRegionName;
+    private final String defaultResourceGroupName;
+    
+    private AzureVolumeOperationSDK operation;
+
+    public AzureVolumePlugin(String confFilePath) {
+        Properties properties = PropertiesUtil.readProperties(confFilePath);
+        this.defaultRegionName = properties.getProperty(AzureConstants.DEFAULT_REGION_NAME_KEY);
+        this.defaultResourceGroupName = properties.getProperty(AzureConstants.DEFAULT_RESOURCE_GROUP_NAME_KEY);
+        this.operation = new AzureVolumeOperationSDK();
+    }
+
+    @Override
+    public boolean isReady(String instanceState) {
+        return AzureStateMapper.map(ResourceType.VOLUME, instanceState).equals(InstanceState.READY);
+    }
+
+    @Override
+    public boolean hasFailed(String instanceState) {
+        return AzureStateMapper.map(ResourceType.VOLUME, instanceState).equals(InstanceState.FAILED);
+    }
+
+    @Override
+    public String requestInstance(VolumeOrder volumeOrder, AzureUser azureUser) throws FogbowException {
```

Comment:
How long takes the Azure Disk operation in the cloud? Is it possible to cause the False Failed Order problem? Do you remember the bug? Talk with me in the Mattermost.

Reply:
The volume or simply _"disk"_ in the context of the Azure cloud, is one of the features that take less time to get ready. But I believe that the asynchronous plugin strategy is still the best even in this case. In system tests carried out in the cloud I did not have any problem of order failure, using either the local region _"brazilsouth"_ or the default _"eastus"_.